Resource Database: Manually deleted resources not handled gracefully

### What happened?

**Steps to Reproduce:**

1. Create a database using terraform
2. Delete database from UI or via cli
3. Run `terraform refresh`

**Expected:**

Refresh succeeds and resources removed from the state.

**Actual:**

Refresh failed with the following logs:

```
planetscale_database.example: Refreshing state... [id=example]
2022-05-11T15:47:27.778+0300 [WARN]  unexpected data: registry.terraform.io/s1ntaxe770r/planetscale:stderr="{"@caller":"github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/internal/logging/helper_schema.go:21","@level":"trace","@message":"Calling downstream","@module":"sdk.helper_schema","@timestamp":"2022-05-11T15:47:27.777984+03:00"}"
2022-05-11T15:47:33.700+0300 [WARN]  unexpected data: registry.terraform.io/s1ntaxe770r/planetscale:stderr="{"@caller":"github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/internal/logging/helper_schema.go:21","@level":"trace","@message":"Called downstream","@module":"sdk.helper_schema","@timestamp":"2022-05-11T15:47:33.700477+03:00"}"
2022-05-11T15:47:33.701+0300 [ERROR] vertex "planetscale_database.example" error: Not Found
2022-05-11T15:47:33.701+0300 [ERROR] vertex "planetscale_database.example" error: Not Found
2022-05-11T15:47:33.701+0300 [ERROR] vertex "planetscale_database.example (expand)" error: Not Found
2022-05-11T15:47:33.701+0300 [INFO]  backend/local: refresh calling Refresh
╷
│ Error: Not Found
│
│   with planetscale_database.example,
│   on main.tf.json line 16, in resource.planetscale_database.example:
│   16:             }
│
╵
2022-05-11T15:47:33.703+0300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"

```

**The root cause seems to be the following:**

From [Hashicorp documentation](https://learn.hashicorp.com/tutorials/terraform/provider-setup#implement-read):

> When you create something in Terraform but delete it manually, Terraform should gracefully handle it. If the API returns an error when the resource doesn't exist, the read function should check to see if the resource is available first. If the resource isn't available, the function should set the ID to an empty string so Terraform "destroys" the resource in state. The following code snippet is an example of how this can be implemented; you do not need to add this to your configuration for this tutorial.
> 
> ```
> if resourceDoesntExist {
>   d.SetID("")
>   return
> }
> ```

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 # Unofficial Terraform Provider for Planetscale :rocket:
 
 - Website: https://planetscale.com
-- Documentation: https://registry.terraform.io/providers/s1ntaxe770r/planetscale/lates
+- Documentation: https://registry.terraform.io/providers/s1ntaxe770r/planetscale/latest
 
 ## Requirements
 

--- a/planetscale/resource_branch.go
+++ b/planetscale/resource_branch.go
@@ -161,6 +161,11 @@ func resourceBranchRead(d *schema.ResourceData, m interface{}) error {
 		Branch:       "",
 	})
 	if err != nil {
+		// Note(turkenh): Handle "Not Found" gracefully
+		if psErr, ok := err.(*ps.Error); ok && psErr.Code == ps.ErrNotFound {
+			d.SetId("")
+			return nil
+		}
 		return errors.New(err.Error())
 	}
 	if err := d.Set("branch", flattenBranch(branchReq)); err != nil {

--- a/planetscale/resource_branch_password.go
+++ b/planetscale/resource_branch_password.go
@@ -180,6 +180,11 @@ func resourceBranchPasswordRead(d *schema.ResourceData, m interface{}) error {
 		PasswordId:   id.(string),
 	})
 	if err != nil {
+		// Note(turkenh): Handle "Not Found" gracefully
+		if psErr, ok := err.(*ps.Error); ok && psErr.Code == ps.ErrNotFound {
+			d.SetId("")
+			return nil
+		}
 		return errors.New(err.Error())
 	}
 

--- a/planetscale/resource_database.go
+++ b/planetscale/resource_database.go
@@ -120,6 +120,18 @@ func resourceDatabaseRead(d *schema.ResourceData, m interface{}) error {
 		Database:     db.(string),
 	})
 	if err != nil {
+		// Note(turkenh): Handle "Not Found" gracefully
+		// From https://learn.hashicorp.com/tutorials/terraform/provider-setup#implement-read
+		// > When you create something in Terraform but delete it manually,
+		// > Terraform should gracefully handle it. If the API returns an error
+		// > when the resource doesn't exist, the read function should check to
+		// > see if the resource is available first. If the resource isn't
+		// > available, the function should set the ID to an empty string so
+		// > Terraform "destroys" the resource in state.
+		if psErr, ok := err.(*ps.Error); ok && psErr.Code == ps.ErrNotFound {
+			d.SetId("")
+			return nil
+		}
 		return errors.New(err.Error())
 	}
 	if err := d.Set("database", flattenDatabase(databaseresp)); err != nil {

